src/Model/common.hpp

```diff
@@ -119,115 +130,131 @@ void seissol::model::getTransposedElasticCoefficientMatrix( seissol::model::Elas
-  std::array<int, 3> velocity_indices = {6,7,8};
-  using Matrix33 = Eigen::Matrix<real, 3, 3>;
-  Matrix33 R11 = R(traction_indices, {0,1,2});
-  Matrix33 R21 = R(velocity_indices, {0,1,2});
-  auto S = - (R21 * R11.inverse()).eval();
-
-  //set lower left block
-  int row = 0;
-  for (auto &t: traction_indices) {
-    int col = 0;
-    for (auto &v: velocity_indices) {
-      QgodLocal(t, v) = S(row, col);
-      col++;
+  if (testIfElastic(material.mu)) {
+    std::array<int, 3> traction_indices = {0,3,5};
+    std::array<int, 3> velocity_indices = {6,7,8};
+    using Matrix33 = Eigen::Matrix<real, 3, 3>;
+    Matrix33 R11 = R(traction_indices, {0,1,2});
+    Matrix33 R21 = R(velocity_indices, {0,1,2});
+    auto S = - (R21 * R11.inverse()).eval();
+
+    //set lower left block
+    int row = 0;
+    for (auto &t: traction_indices) {
+      int col = 0;
+      for (auto &v: velocity_indices) {
+        QgodLocal(t, v) = S(row, col);
+        col++;
+      }
+      row++;
     }
-    row++;
-  }
-  //set lower right block
-  for (auto &v : velocity_indices) {
-    QgodLocal(v, v) = 1.0;
+    //set lower right block
+    for (auto &v : velocity_indices) {
+      QgodLocal(v, v) = 1.0;
+    }
+  } else {
+    // Acoustic material only has one traction (=pressure) and one velocity comp.
+    // relevant to the Riemann problem
+    QgodLocal(0, 6) = -1 * R(6,0) * 1/R(0,0); // S
+    QgodLocal(6, 6) = 1.0;
   }
 }
 
+
 template<typename Tloc, typename Tneigh>
-void seissol::model::getTransposedElasticGodunovState( Material const&                      local,
-    Material const&                      neighbor,
-    enum ::faceType                      faceType,
-    Tloc&                                QgodLocal,
-    Tneigh&                              QgodNeighbor )
-{
+void seissol::model::getTransposedElasticGodunovState(Material const& local,
+                                                      Material const& neighbor,
+                                                      FaceType faceType,
+                                                      Tloc& QgodLocal,
+                                                      Tneigh& QgodNeighbor) {
   QgodNeighbor.setZero();
 
+  // Eigenvectors are precomputed
   Matrix99 R = Matrix99::Zero();
 
-  //eigenvectors have been precalculated
-  R(0,0) = local.lambda + 2*local.mu;
-  R(0,8) = neighbor.lambda + 2*neighbor.mu;
-  R(1,0) = local.lambda;
+  if (testIfElastic(local.mu)) {
+    R(0,0) = local.lambda + 2*local.mu;
+    R(1,0) = local.lambda;
+    R(2,0) = local.lambda;
+    R(6, 0) = std::sqrt((local.lambda + 2 * local.mu) / local.rho);
+
+    R(3,1) = local.mu;
+    R(7,1) = std::sqrt(local.mu / local.rho);
+
+    R(5,2) = local.mu;
+    R(8,2) = std::sqrt(local.mu / local.rho);
+
+  } else {
+    R(0,0) = local.lambda;
+    R(1,0) = local.lambda;
+    R(2,0) = local.lambda;
+    R(6,0) = std::sqrt((local.lambda + 2 * local.mu) / local.rho);
+
+    R(3,1) = 1.0;
+    R(5,2) = 1.0;
+  }
```

Comment:
Isnt this if statement just duplicated code? In the end, if the material is not elastic, you would just add an 2*epsilon.